### PR TITLE
Fix issue 181

### LINKED ISSUE #181: Refactor and move all `cond*_` functions

### PR DESCRIPTION
**Pull request by SarahAlidoost**
### Description
closes #181, the merge is pending addressing issues mentioned [here](https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/181#issuecomment-1658570621). This branch re-produces the same results as those generated by the main branch, tested with a small dataset.

### Checklist
- [x] Add a reference to related issues.
- [x] @mentions of the person or team responsible for reviewing proposed changes.
- [x] This pull request has a descriptive title.
- [x] Code is written according to the [guidelines](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf).
- [x] The checks by [MISS_HIT style checker and linter](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/CONTRIBUTING.md#follow-matlab-style-guidelines), below the pull request, are successful (green).
- [x] Documentation is available.
- [x] Add changes to the [changelog file](CHANGELOG.md) under section `Unreleased`.
- [x] Model runs successfully.
- [x] Ask a meinatainer to re-generate exe file if matlab codes are changed. About how to create an exe file, see [exe readme](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/run_model_on_snellius/exe/README.md).

### CODE REVIEW

**Review by BSchilperoort — APPROVED** [review]
Looks good, nice work! I checked the model results and there seems to be no change at all between `main` and this branch.

It would be nice to have docstrings in many of the new functions though (what does the function do? what are the inputs? what's the source of the equations?). But I know you can't write those. Perhaps keeping track of the functions that still need a docstring in an issue could be a good idea?

File: src/+conductivity/+hydraulicConductivity/calculateDTheta.m [hunk 1]
```diff
@@ -0,0 +1,43 @@
+function dtheta = calculateDTheta(subRoutine, heat_term, theta_s, theta_r, theta_m, gama_hh, theta_ll, theta_l, theta_uu, theta_u, hh, h, hh_frez, h_frez, phi_s, lamda, alpha, n, m)
+
+    % get soil constants
+    SoilConstants = io.getSoilConstants();
+
+    switch subRoutine
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply src/+conductivity/+hydraulicConductivity/calculateDTheta.m:6]: Issue #192 submitted.

File: src/+conductivity/+hydraulicConductivity/calculateGama_hh.m [hunk 2]
```diff
@@ -0,0 +1,12 @@
+function gama_hh = calculateGama_hh(hh)
```
[the remark this answers is not in the document]
Comment (SarahAlidoost) [reply src/+conductivity/+hydraulicConductivity/calculateGama_hh.m]: Right, I fixed the typo in `Gama_h*` and `gama_h* `

**Comment by SarahAlidoost** [comment]
> Looks good, nice work! I checked the model results and there seems to be no change at all between `main` and this branch.
> 
> It would be nice to have docstrings in many of the new functions though (what does the function do? what are the inputs? what's the source of the equations?). But I know you can't write those. Perhaps keeping track of the functions that still need a docstring in an issue could be a good idea?

thanks a lot for reviewing and running the code. I agree that the code needs more documentation, I submitted issue #191 for this.

**Comment by SarahAlidoost** [comment]
Waiting for addressing issues mentioned [here](https://github.com/EcoExtreML/STEMMUS_SCOPE/issues/181#issuecomment-1658570621) before the merge.

**Pull request merged**